nexus/api.go

```diff
@@ -145,6 +146,25 @@ func addCallbackHeaderToHTTPHeader(nexusHeader Header, httpHeader http.Header) h
 	return httpHeader
 }
 
+func addLinksToHTTPHeader(links []Link, httpHeader http.Header) error {
```

Comment:
No need to model the non-exported function as returning an error if it never does (refactoring is safe to add it later if needed). Having said that, this should return an error because `encodeLink` should return an error on invalid link.